.expeditor/scripts/release_habitat/package_and_upload_binary.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+# Unpack the hart file from our channel, repack it, and upload it to
+# package-router
+
+set -euo pipefail
+
+source .expeditor/scripts/release_habitat/shared.sh
+
+export HAB_AUTH_TOKEN="${ACCEPTANCE_HAB_AUTH_TOKEN}"
+export HAB_BLDR_URL="${ACCEPTANCE_HAB_BLDR_URL}"
+
+channel=$(get_release_channel)
+
+echo "--- Channel: $channel - bldr url: $HAB_BLDR_URL"
+
+release_version=$(get_latest_pkg_release_version_in_release_channel "hab")
+
+declare -g hab_binary
+install_release_channel_hab_binary "x86_64-linux"
+
+${hab_binary} pkg install core/gzip \
+  core/tar \
+  core/wget \
+  core/zip
+
+# Import packages@chef.io GPG signing key
+aws s3 cp s3://chef-cd-citadel/packages_at_chef.io.pgp \
+  packages_at_chef.io.pgp \
+  --profile=chef-cd 
+gpg --import packages_at_chef.io.pgp
+
+tmp_root="$(mktemp -d -t "grant-XXXX")"
+extract_dir="$tmp_root/extract"
+mkdir -p "$extract_dir"
+
+echo "--- Downloading $release_version for $BUILD_PKG_TARGET"
+if [[ $BUILD_PKG_TARGET == *"darwin" ]]; then
+  artifact_name=$(buildkite-agent meta-data get MACOS_ARTIFACT)
+  buildkite-agent artifact download "$artifact_name" "$tmp_root/" --step "[:macos: build hab]"
+  mv "$tmp_root/$artifact_name" "$tmp_root/hab-$channel.hart"
+else
+  # once https://github.com/habitat-sh/habitat/issues/6878 is released,
+  # we can use hab pkg download rather than this wget
+  hab pkg exec core/wget wget \
+    "${HAB_BLDR_URL}/v1/depot/pkgs/core/hab/$release_version/download?target=$BUILD_PKG_TARGET" \
+    -O "$tmp_root/hab-$channel.hart"
+fi
+
+target_hart="$tmp_root/hab-$channel.hart"
+if [[ $(head -n 1 "${target_hart}") != HART-1 ]]; then
+  echo "Hart file does not match expected format, exiting."
+  exit 1
+fi
+tail -n+6 "${target_hart}" | \
+    tar --directory "${extract_dir}" \
+        --extract \
+        --xz \
+        --strip-components=6
+
+extracted_hab_binary="$(find "$extract_dir" \( -name hab -or -name hab.exe \) -type f)"
+pkg_ident="$(tr --delete '\r' < "$extract_dir"/IDENT)"
+pkg_version="$(echo "$pkg_ident" | cut -d '/' -f 3)"
+pkg_release="$(echo "$pkg_ident" | cut -d '/' -f 4)"
+archive_name="hab-${pkg_version}-${pkg_release}-$BUILD_PKG_TARGET"
+
+build_dir="$tmp_root/build"
+pkg_dir="$build_dir/${archive_name}"
+
+echo "Copying $extracted_hab_binary to $(basename "$pkg_dir")"
+mkdir -p "$pkg_dir"
+mkdir -p "$tmp_root/results"
+
+if [[ $BUILD_PKG_TARGET == *"windows" ]]; then
+  # windows has more than just the hab binary, so we need to copy all the files
+  # in the directory
+  for file in "$(dirname "$extracted_hab_binary")"/*; do 
+    cp -p "$file" "$pkg_dir/"
+  done
+else
+  cp -p "$extracted_hab_binary" "$pkg_dir/$(basename "$extracted_hab_binary")"
+fi
+
+echo "Compressing 'hab' binary"
+(cd "$build_dir"
+  case "$BUILD_PKG_TARGET" in
+  *-linux | *-linux-kernel2)
+      pkg_artifact="$tmp_root/results/${archive_name}.tar.gz"
+      tarball="$build_dir/$(basename "${pkg_artifact}.gz")"
```

Comment:
This won't strip the `.gz`; the suggestion I made in https://github.com/habitat-sh/habitat/pull/6908#discussion_r321530190 was slightly different:
```suggestion
      tarball="$build_dir/$(basename "${pkg_artifact} .gz")"
```